### DISCUSSION TITLE
Theme generator based on user input

### DISCUSSION THREAD

**Discussion by petermazzocco** [saadeghi/daisyui#2320] [Q&A] [answered]
Hello!

I was wonder if there is a way to implement the Theme Generator (https://daisyui.com/theme-generator/) where it can update the theme based on inputted hex codes from a user? 

I would like to incorporate that into my next app in the `tailwind.config.js` file!

**Comment by saadeghi** [comment, marked as answer]
Theme colors are HSL values in CSS variables: https://unpkg.com/browse/daisyui@3.6.4/dist/themes.css

If you want to get Hex values from the user (something like daisyUI Theme Generator), You need to convert hex to HSL, using JS. I used [colord](https://github.com/omgovich/colord) and [here's the source code for daisyUI theme generator](https://github.com/saadeghi/daisyui/blob/master/src/docs/src/routes/theme-generator/%2Bpage.svelte)

Then you can set HSL values for each CSS variable.

Let me know if you have any questions.

**Reply by petermazzocco** [reply]
Thank you! These are very useful and I will get these implemented asap!
